### DISCUSSION TITLE
UHI+UWG Questions

### DISCUSSION THREAD

**Discussion by AlexKDo** [UMEP-dev/UMEP#508] [Q&A]
Hello again.. I am trying to run UHI Feature now and I am having some trouble.. 
So I did:

3.16. Urban Land Cover: Land Cover Fraction (Grid)
3.18. Urban Land Cover: Land Cover Reclassifier
3.19. Urban Morphology: Morphometric Calculator (Grid)

and I am having trouble with:

3.14. Urban Heat Island: UWG Prepare
4.10. Urban Heat Island: Urban Weather Generator

In UWG Prepare I should use the vector polygon grid that depicts the modelling area(s). This should be the same polygon layer that was used in  Urban Weather Generator. But in Generator I should have the parth that contains .uwg-files generated by UWGPrepare.But I dont have it since I cannot seccessfully run Prepare.

extra Question: Grid files should be the same vector polygon grid or 1 for 3.16 and a new one for 3.19?

**Comment by biglimp** [comment]
You should do LandCoverReclassifier before you use the LandCoverFraction (Grid). The Grid you use should be the same for all tools in this process.

**Reply by AlexKDo** [reply]
So I did in this order and I did a first run but I had many errors+ when I tried UWG Analyzer I pressed Generate many times but nothing happened. So this is an example of error. I think something went wrong with the previous steps I did.. This is a random PrtSc. 
![Στιγμιότυπο οθόνης 2023-03-18 010719](https://user-images.githubusercontent.com/81244161/226068334-d9fec107-84fc-42bf-92d1-81265190cf17.png)

**Reply by biglimp** [reply]
So, open the Python Console in QGIS, rerun UWG and report the error displayed fro one grid.

**Reply by AlexKDo** [reply]
so I did this to see the error and I cancelled the generator. This is the msg

Traceback error message while caclulation grid: 168
Traceback (most recent call last):
  File "C:\Users/alexk/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\processor\uwg_algorithm.py", line 224, in processAlgorithm
    model.generate()
  File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\uwg.py", line 1176, in generate
    self._compute_input()
  File "C:\OSGeo4W\apps\Python39\lib\site-packages\uwg\uwg.py", line 1671, in _compute_input
    road_veg_coverage = self.vegcover / (1 - self.blddensity)
ZeroDivisionError: float division by zero

**Comment by biglimp** [comment]
So it looks like you have building density = 1 (nothing but buildings) which the model so not accept. Check you input for this grid and see if that is true.

**Reply by AlexKDo** [reply]
Yes, from LandCoverage Grid I have 1 and from Morphology I have 1 too. And this is the same for all grids that are buildings.

**Reply by biglimp** [reply]
Then you must have something wrong when you calculated fractions and morphology. Can you attach two sceendumps from one grid in QGIS, one showing land cover and the other one showing your DSM.

**Reply by AlexKDo** [reply]
I have them in the same grid layer. This is an example

![image](https://user-images.githubusercontent.com/81244161/226338270-92e0bd1a-ce18-4b66-b404-21e239e2424d.png)

**Reply by biglimp** [reply]
No, I want to see your raster data in your map canvas.

**Comment by AlexKDo** [comment]
![image](https://user-images.githubusercontent.com/81244161/226338969-8224279e-54b2-4253-bd8b-a561ec0df46f.png)

![image](https://user-images.githubusercontent.com/81244161/226339074-e10f0637-804c-4832-9fc3-a0ad964f1863.png)

**Reply by biglimp** [reply]
Thanks. 
UWG is a local scale model so your grids are too small. I recommend using grids that are at least 200x200 meters. Even larger are recommended (400x400).

**Reply by AlexKDo** [reply]
Ok. Thank you very much!
